DESIGN.md

```diff
@@ -0,0 +1,56 @@
+# TLS Provider Design
+
+The TLS Provider offers a small surface area compared to other providers (like
+[AWS](https://registry.terraform.io/providers/hashicorp/aws/latest),
+[Google](https://registry.terraform.io/providers/hashicorp/google/latest),
+[Azure](https://registry.terraform.io/providers/hashicorp/azurerm/latest), ...),
+and focuses on covering the needs of working with entities like
+keys and certificates, that are part of
+[Transport Security Layer](https://en.wikipedia.org/wiki/Transport_Layer_Security).
+
+Below we have a collection of _Goals_, _Patterns_ and _Taboos_: they represent the guiding principles applied during
+the development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+ 
+## Goals
+
+* Support [cryptography](https://en.wikipedia.org/wiki/Cryptography) _primitives_ necessary to Terraform configurations
+* Provide managed resourced and data sources to manipulate and interact with **Keys, Certificates and Certificate Requests**
+* Support file and data formats widely used in the industry: this means sticking with open and well documented standards
+* Support cryptography key algorithms in line with widely used software
```

Comment:
I like the sentiment here, although this also feels very broad and open to interpretation. Maybe this could be clarified to list out official references the current digital signature key algorithms supported today then offer a quick sentence or two above/below all these about how to submit design change proposals (e.g. feature requests).

Reply:
Same as above. Yeah, my first pass was WAY too vague and open to interpretation.